packages/app-store/zoho-bigin/lib/CalendarService.ts

```diff
@@ -0,0 +1,312 @@
+      return {
+        First_Name: firstName,
+        Last_Name: lastName,
+        Email: attendee.email,
+      };
+    });
+
+    return axios({
+      method: "post",
+      url: token.api_domain + this.contactsSlug,
+      headers: {
+        "content-type": "application/json",
+        authorization: `Zoho-oauthtoken ${token.access_token}`,
+      },
+      data: JSON.stringify({ data: contacts }),
+    });
+  }
+
+  /***
+   * Finds existing Zoho Bigin Contact record based on email address. Returns a list of contacts objects that matched.
+   */
+  private async contactSearch(event: CalendarEvent) {
+    const token = await this.auth.getToken();
+    const searchCriteria =
+      "(" + event.attendees.map((attendee) => `(Email:equals:${encodeURI(attendee.email)})`).join("or") + ")";
+
+    return await axios({
+      method: "get",
+      url: `${token.api_domain}${this.contactsSlug}/search?criteria=${searchCriteria}`,
+      headers: {
+        authorization: `Zoho-oauthtoken ${token.access_token}`,
+      },
+    })
+      .then((data) => data.data)
+      .catch((e) => this.log.error("Error searching contact:", JSON.stringify(e), e.response?.data));
+  }
+
+  /***
+   * Sends request to Zoho Bigin API to add new Events.
+   */
+  private async createBiginEvent(event: CalendarEvent) {
+    const token = await this.auth.getToken();
+    const biginEvent = {
+      Event_Title: event.title,
+      Start_DateTime: toISO8601String(new Date(event.startTime)),
+      End_DateTime: toISO8601String(new Date(event.endTime)),
+      Description: event.additionalNotes,
+      Location: getLocation(event),
+    };
+
+    return axios({
+      method: "post",
+      url: token.api_domain + this.eventsSlug,
+      headers: {
+        "content-type": "application/json",
+        authorization: `Zoho-oauthtoken ${token.access_token}`,
+      },
+      data: JSON.stringify({ data: [biginEvent] }),
+    })
+      .then((data) => data.data)
+      .catch((e) => this.log.error("Error creating bigin event", JSON.stringify(e), e.response?.data));
+  }
+
+  /***
+   * Handles orchestrating the creation of new events in Zoho Bigin.
+   */
+  async handleEventCreation(event: CalendarEvent, contacts: CalendarEvent["attendees"]) {
+    const meetingEvent = await this.createBiginEvent(event);
+    if (meetingEvent.data && meetingEvent.data.length && meetingEvent.data[0].status === "success") {
+      this.log.debug("event:creation:ok", { meetingEvent });
+      return Promise.resolve({
+        uid: meetingEvent.data[0].details.id,
+        id: meetingEvent.data[0].details.id,
+        //FIXME: `externalCalendarId` is required by the `updateAllCalendarEvents` method, but is not used by zoho-bigin App. Not setting this property actually skips calling updateEvent..
+        // Here the value doesn't matter. We just need to set it to something.
+        externalCalendarId: "NO_CALENDAR_ID_NEEDED",
```

Comment:
This was causing reschedule to not update the Bigin Event. This might be a problem with other CRM apps as well.